apis/v1alpha1/gateway_types.go

```diff
@@ -252,8 +252,8 @@ type Listener struct {
 // Implementations can define their own protocols if a core ProtocolType does not
 // exist. Such definitions must use prefixed name, such as
 // `mycompany.com/my-custom-protocol`. Un-prefixed names are reserved for core
-// protocols. Any protocol defined by implementations will fall under custom
-// conformance.
+// and extended protocols. Any protocol defined by implementations will fall
```

Comment:
What are extended protocols here? It looks like we haven't associated conformance levels with the protocols defined below.

Reply:
Good question. The protocol themselves will probably not make sense to have conformance levels defined. I think conformance should be based on how the Protocol is being used, for example, for Route associated with the protocol. Like, there have been brief statements saying that TCP and UDP routes would be extended conformance in the beginning and then later be moved into core as we gain experience from implementers.

This begs the question: Why should protocols be tied to conformance ladder? I feel like I could be missing something.

cc @danehans @bowei

Reply:
If I'm remembering this correctly, this all came from the idea that we wanted to allow users to specify protocols that we're not defined within this API. Previously we had strict validation that protocol must come from the list defined here. That really limited the extensibility of the API. I think what we're trying to communicate here is if the protocol is not defined in the API, a domain prefixed path should be used to ensure it does not conflict with any future protocols added to the API.

Reply:
If I were to implement `http3` or similar, it feels weird to put `mycompany.com/http3`, even if it later becomes part of the API. This would just make users have to migrate from  `mycompany.com/http3` to `http3` if it does become core.

Reply:
Yeah, that's a good point. Here's the context for the original decision around this: https://github.com/kubernetes-sigs/service-apis/pull/398#discussion_r510493532. It's hard to find a good list to act as a reference here so at this point we've just said we'll expand our list as additional commonly used protocols are supported. That will admittedly not scale well, so any alternative solutions are very welcome here.

Reply:
A bit more discussion around this in https://github.com/kubernetes-sigs/service-apis/issues/433.

Reply:
> If I were to implement http3 or similar, it feels weird to put mycompany.com/http3, even if it later becomes part of the API. This would just make users have to migrate from mycompany.com/http3 to http3 if it does become core.

What is really the problem with that? I see that as completely fine and migration to be gradual and painless too.